be/src/connector/mysql_connector.cpp

```diff
@@ -0,0 +1,474 @@
+                APPEND_TO_SQL                                                                 \
+            }                                                                                 \
+            filters_in.emplace(field_name, vector_values);                                    \
+        }                                                                                     \
+        break;                                                                                \
+    }
+
+#define DIRECT_APPEND_TO_SQL vector_values.emplace_back(std::to_string(value));
+                    APPLY_FOR_NUMERICAL_TYPE(READ_CONST_PREDICATE, DIRECT_APPEND_TO_SQL)
+#undef APPLY_FOR_NUMERICAL_TYPE
+#undef DIRECT_APPEND_TO_SQL
+
+#define CONVERT_APPEND_TO_SQL          \
+    std::stringstream ss;              \
+    for (char c : value.to_string()) { \
+        if (c == '"') {                \
+            ss << '\\';                \
+        }                              \
+        ss << c;                       \
+    }                                  \
+    vector_values.emplace_back(fmt::format("'{}'", ss.str()));
+                    APPLY_FOR_VARCHAR_DATE_TYPE(READ_CONST_PREDICATE, CONVERT_APPEND_TO_SQL)
+#undef APPLY_FOR_VARCHAR_DATE_TYPE
+#undef CONVERT_APPEND_TO_SQL
+
+                    case INVALID_TYPE:
+                    case TYPE_NULL:
+                    case TYPE_BINARY:
+                    case TYPE_DECIMAL:
+                    case TYPE_STRUCT:
+                    case TYPE_ARRAY:
+                    case TYPE_MAP:
+                    case TYPE_HLL:
+                    case TYPE_TIME:
+                    case TYPE_OBJECT:
+                    case TYPE_PERCENTILE:
+                    case TYPE_LARGEINT:
+                    case TYPE_DECIMAL128:
+                    case TYPE_DECIMALV2:
+                    case TYPE_DECIMAL32:
+                    case TYPE_DECIMAL64:
+                    case TYPE_DOUBLE:
+                    case TYPE_FLOAT:
+                    case TYPE_JSON:
+                        break;
+                }
+            }
+        }
+    }
+
+    RETURN_IF_ERROR(_mysql_scanner->query(_table_name, _columns, _filters, filters_in, filters_null_in_set, _read_limit));
+    // check materialize slot num
+    int materialize_num = 0;
+
+    for (const auto& slot : _tuple_desc->slots()) {
+        if (slot->is_materialized()) {
+            materialize_num++;
+        }
+    }
+
+    if (_mysql_scanner->field_num() != materialize_num) {
+        return Status::InternalError("input and output not equal.");
+    }
+
+    return Status::OK();
+}
+
+Status MySQLDataSource::get_next(RuntimeState* state, vectorized::ChunkPtr* chunk) {
+    VLOG(1) << "MySQLDataSource::GetNext";
+
+    DCHECK(state != nullptr && chunk != nullptr);
+
+    RETURN_IF_CANCELLED(state);
+    SCOPED_TIMER(_runtime_profile->total_time_counter());
+
+    // TODO: check limit
+    if (_is_finished) {
```

Comment:
No need to maintain this variable.  Return EOF, then get_next will not be called any more.